lgn-messages/src/types/v1/preprocessing/ext_tasks.rs

```diff
@@ -67,17 +71,29 @@ pub enum MptType {
 pub struct MappingLeafInput {
     pub key: Vec<u8>,
     pub node: Vec<u8>,
-    pub slot: usize,
-    pub contract_address: Address,
+    pub slot: u8,
+    pub key_id: u64,
+    pub value_id: u64,
 }
 
 impl MappingLeafInput {
-    pub fn new(key: Vec<u8>, node: Vec<u8>, slot: usize, contract_address: Address) -> Self {
+    pub fn new(
+        key: Vec<u8>,
+        node: Vec<u8>,
+        slot: u8,
+        contract_address: &Address,
+        chain_id: u64,
+    ) -> Self {
+        let key_id = identifier_for_mapping_key_column(slot, contract_address, chain_id, vec![]);
+        let value_id =
+            identifier_for_mapping_value_column(slot, contract_address, chain_id, vec![]);
```

Comment:
We want these identifiers to be passed in. The only place where these identifiers should be computed is in the `api` crate in `distributed-query`.

The flow looks like this:
- The user gets the generated identifiers from the API
- Then registers them on chain
- Then inserted into the `zkmr.table_registrations` table
- Then they should be READ-ONLY from the DB in every other place they are needed

The reasoning is that we want the flexibility to specify whatever data we need in the `extra` field (which is set to `vec![]` here). And we want to do this without having to copy/access the data for `extra` in many places throughout the codebase.

Also, we want the identifier changes to be forwards compatible. Meaning we don't want to have to recompute old ones if we change the inputs to the `extra` field.

Reply:
Ah I see; I thought the goal was to compute them at the QE level, but still live. I'll fix that tomorrow.